src/components/draw/feature-sub-content.vue

```diff
@@ -20,35 +20,51 @@ const emit = defineEmits([
   'toggleEditFeature',
   'toggleDock',
   'clickDelete',
-  'submitEditInfo',
+  'resetInfo',
+  'resetStyle',
+  'submitEditFeature',
 ])
 const { t } = useTranslation()
-const feature: DrawnFeature | undefined = inject('feature')
+const feature: DrawnFeature = inject('feature')!
+let prevLabel = feature.label
+let prevDescription = feature.description
+// keep deep copy of previous style to be able to revert style on cancel
+const prevStyle: DrawnFeatureStyle = { ...feature.featureStyle }
 
 const editComponents = {
   FeatureConcentricCircle,
   FeatureEditInfo,
   FeatureEditStyle,
   FeatureConfirmDelete,
-  FeatureEditSymbol,
 }
 const currentEditCompKey: Ref<keyof typeof editComponents | undefined> =
   ref(undefined)
 
 provide('currentEditCompKey', currentEditCompKey)
 
 function onClickCancel() {
+  if (currentEditCompKey.value == 'FeatureEditStyle') {
+    emit('resetStyle', prevStyle)
+  } else if (currentEditCompKey.value == 'FeatureEditInfo') {
+    emit('resetInfo', prevLabel, prevDescription)
+  }
   currentEditCompKey.value = undefined
 }
 
 function onClickValidate() {
   const currentComponent =
     editComponents[currentEditCompKey.value as keyof typeof editComponents]
 
+  prevLabel = feature.label
+  prevDescription = feature.description
+  Object.assign(prevStyle, feature.featureStyle)
```

Comment:
Object assign?

Reply:
OK

Reply:
ah, I just saw, this writing makes the linter catch a const modification, I see a whole new advantage int the spread syntax.

cool!